app/forms/steps/case/charges_form.rb

```diff
@@ -5,6 +5,8 @@ class ChargesForm < Steps::BaseFormObject
 
       delegate :offence_dates_attributes=, to: :record
 
+      validates_with ChargesValidator
```

Comment:
We will want to also implement `unless: :any_marked_for_destruction?` otherwise you can't delete blank dates if the user mistakenly add another date and then they try to remove it while being blank/invalid.
Same I did for codefendants.

However this can be done in a separate PR, not a problem.